app/models/user_project_role.rb

```diff
@@ -0,0 +1,15 @@
+class UserProjectRole < ActiveRecord::Base
+  include HasProjectRole
+  belongs_to :project
+  belongs_to :user
+
+  validates_presence_of :project, :user
+  validates :role_id, inclusion: { in: ProjectRole.all.map(&:id) }
```

Comment:
`ProjectRole` is an `ActiveHash` (https://github.com/zilkey/active_hash), not an `ActiveRecord`. According to their documentation, `pluck` is not supported, but I'll take a look.

Anyway, good to know that method. Was not aware of what it did.